### DISCUSSION TITLE
How to enable hashing

### DISCUSSION THREAD

**Discussion by Tosheen** [tailwindlabs/tailwindcss#11927] [Help]
Currently I am writing an UI library using React and Tailwind for an internal project. 
It works great however I am wondering if there is a way to hash class names so that there are no potential issues with conflicting classes.

Lets say this UI library will be installed inside a project which is already using Tailwind.
I am aware of the prefix rule but I was hoping that there is another solution which will convert class name like bg-blue-500 into something like cs-1sbzy4 for example.
Does tailwind support that or are there any plans in the future to support it?

Solutions like twind and windi are dead. 
Unocss doesnt support hashing.
Twin.Macro does support it and it pretty much on par with v3 however syntax is just not appealing and I do not want to use another CSS-in-JS solution.

Any recommendations? What would be the best approach to develop an UI library using Tailwind.css

Thank you.

**Comment by NotJustClarkKent** [comment]
We're in the same boat where we have some internal projects that already use Tailwind but we want to write a reusable component library to be used a long side also leveraging Tailwind and we don't want them to collide.

I have a hacky/poc solution working where I create an external css file, import that into the component, then use the `@apply` directive in the css file. This wont scale well and does not handle custom properties (ie: `--tw-bg-opacity`) so I'm hoping I can figure out something in a build step.

```css
// button.css
.button {
    @apply bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded;
}
```

```js
// Button.tsx
import styles from './button.css';
...
<button className={styles}>Test</button>
```
 If anyone has any hints or ideas, I'm all ears. 😄 🙏

**Reply by RenovZ** [reply]
Can we have a better solution, without defining a extra button class, which may can be processed automatically after the tailwind compiler.
